ios/lib/ReactNativeCameraKit/CKCamera.m

```diff
@@ -506,96 +506,81 @@ - (void)snapStillImage:(BOOL)shouldSaveToCameraRoll success:(CaptureBlock)block
     dispatch_async( self.sessionQueue, ^{
         AVCaptureConnection *connection = [self.stillImageOutput connectionWithMediaType:AVMediaTypeVideo];
         
-        // Update the orientation on the still image output video connection before capturing.
-        connection.videoOrientation = self.previewLayer.connection.videoOrientation;
-        
+        UIImageOrientation imageOrientation = UIImageOrientationUp;
+        switch([UIDevice currentDevice].orientation) {
+            default:
+            case UIDeviceOrientationPortrait:
+                connection.videoOrientation = AVCaptureVideoOrientationPortrait;
+                imageOrientation = UIImageOrientationUp;
+                break;
+            case UIDeviceOrientationPortraitUpsideDown:
+                connection.videoOrientation = AVCaptureVideoOrientationPortraitUpsideDown;
+                imageOrientation = UIImageOrientationDown;
+                break;
+            case UIDeviceOrientationLandscapeLeft:
+                imageOrientation = UIImageOrientationRight;
+                connection.videoOrientation = AVCaptureVideoOrientationLandscapeRight;
+                break;
+            case UIDeviceOrientationLandscapeRight:
+                connection.videoOrientation = AVCaptureVideoOrientationLandscapeLeft;
+                imageOrientation = UIImageOrientationRightMirrored;
+                break;
+        }
         
         // Capture a still image.
         [self.stillImageOutput captureStillImageAsynchronouslyFromConnection:connection completionHandler:^( CMSampleBufferRef imageDataSampleBuffer, NSError *error ) {
-            if ( imageDataSampleBuffer ) {
-                // The sample buffer is not retained. Create image data before saving the still image to the photo library asynchronously.
-                NSData *imageData = [AVCaptureStillImageOutput jpegStillImageNSDataRepresentation:imageDataSampleBuffer];
-                UIImage *capturedImage = [UIImage imageWithData:imageData];
-                capturedImage = [CKCamera rotateImage:capturedImage];
-                
-                CGSize previewScaleSize = [CKCamera cropImageToPreviewSize:capturedImage size:self.previewLayer.bounds.size];
-                CGRect rectToCrop = CGRectMake((capturedImage.size.width-previewScaleSize.width)*0.5, (capturedImage.size.height-previewScaleSize.height)*0.5, previewScaleSize.width, previewScaleSize.height);
-                
-                if (self.ratioOverlayString) {
+            if (!imageDataSampleBuffer) {
+                //NSLog( @"Could not capture still image: %@", error );
```

Comment:
Should we uncomment this line and log the error ?

Reply:
Yup.